docusaurus/docs/introduction/backend.md

```diff
@@ -83,18 +79,30 @@ Examples of use cases for implementing resources:
 - Use [chunked transfer encoding](https://en.wikipedia.org/wiki/Chunked_transfer_encoding) to return large data responses in chunks or to enable certain streaming capabilities.
 
-#### Health checks
+### Health checks
 
 The health checks capability allows a backend plugin to return the status of the plugin. For data source backend plugins, the health check is automatically called when a user edits a data source and selects _Save & Test_ in the UI.
 
 A plugin's health check endpoint is exposed in the Grafana HTTP API and allows external systems to continuously poll the plugin's health to make sure that it's running and working as expected.
 
-#### Collect metrics
+### Collect metrics
 
 A backend plugin can collect and return runtime, process, and custom metrics using the text-based Prometheus [exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/). If you're using the [Grafana Plugin SDK for Go](./grafana-plugin-sdk-for-go.md) to implement your backend plugin, then the [Prometheus instrumentation library for Go applications](https://github.com/prometheus/client_golang) is built-in. This SDK gives you Go runtime metrics and process metrics out of the box. You can use the [Prometheus instrumentation library](https://github.com/prometheus/client_golang) to add custom metrics to instrument your backend plugin.
 
 The Grafana HTTP API offers an endpoint (`/api/plugins/<plugin id>/metrics`) that allows you to configure a Prometheus instance to scrape the metrics.
 
-#### Streaming
+### Streaming
+
+The streaming capability allows a backend plugin to handle data source queries that are streaming. For more information, refer to [Build a streaming data source plugin](https://grafana.com/docs/grafana/latest/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-streaming-data-source-plugin/).
+
+## Data communication model
+
+When communicating with backend plugins, Grafana should provide all the necessary information (configuration) in each request to allow the plugin to fulfill the request and return a response. This is a simple model for ensuring data communication between Grafana and plugin. 
+
+However, there are cases when a Grafana plugin or data source has been configured with sensitive information that is stored encrypted in Grafana’s database. In such cases, Grafana will decrypt the sensitive data and attach it in clear text in messages to the plugin upon each request to their respective backends.
+
+## Caching and connection pooling
+
+Grafana uses instance management (that is, caching) in the backend plugin SDK to optimize plugin resources. It works by caching parts of the plugin (including `jsonData` and `secureJSONData`) in memory so that subsequent requests can benefit from not having to reinitialize the plugin instance, where the instance is intended to hold things such as HTTP clients, connection pools, decrypted secrets, and so on. 
```

Comment:
```suggestion
Grafana uses instance management in the backend plugin SDK to optimize plugin resources. It works by caching parts of the plugin (including `jsonData` and `secureJSONData`) in memory so that subsequent requests can benefit from not having to reinitialize the plugin instance, where the instance is intended to hold things such as HTTP clients, connection pools, decrypted secrets, and so on. 
```

Reply:
change accepted.